File: frontend/src/types/license.ts
export interface LicenseInfo {
  id: number;
  license_id: string;
  plan_code: string;
  plan_name: string;
  organization_name: string | null;
  issued_at: string;
  expires_at: string;
  max_users: number | null;
  max_armada: number | null;
  features: string[];
  is_active: boolean;
  activated_at: string | null;
  created_at: string;
}

export interface LicenseStatusResponse {
  activated: boolean;
  license: LicenseInfo | null;
}

export interface LicenseActivateRequest {
  license_key: string;
}

File: backend/app/schemas/license.py
from datetime import datetime

from pydantic import BaseModel, Field


class LicenseActivateRequest(BaseModel):
    license_key: str = Field(
        min_length=10,
        max_length=2048,
    )


class LicenseResponse(BaseModel):
    id: int
    license_id: str
    plan_code: str
    plan_name: str
    organization_name: str | None = None
    issued_at: datetime
    expires_at: datetime
    max_users: int | None = None
    max_armada: int | None = None
    features: list[str]
    is_active: bool
    activated_at: datetime | None = None
    created_at: datetime

    class Config:
        from_attributes = True


class LicenseStatusResponse(BaseModel):
    activated: bool
    license: LicenseResponse | None = None


class LicenseGenerateRequest(BaseModel):
    plan_code: str
    organization_name: str
    years: int = 1


class LicenseGenerateResponse(BaseModel):
    license_key: str
    license_id: str
    expires_at: str